### PR TITLE
[ENH] automated code generation with context7 MCP server and rules

### PR DESCRIPTION
**Pull request by moyuanchen**
#### What does this implement/fix? Explain your changes.

Added mcp server configuration file and context7 rule file for using cursor agent to automatically generate a portfolio for the user when prompted. 

Some examples of usage in agent mode and plan mode is included.

#### Did you add any tests for the change?

See examples/mcp/

##### For all contributions
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG]. [BUG] - bugfix, [MNT] - CI, test framework, [ENH] - adding or improving code, [DOC] - writing or improving documentation or docstrings.

##### For new estimators
- [ ] I've added the estimator to the API reference in `docs/api.rst`.
- [ ] I've added one or more illustrative usage examples to the docstring and the `examples` section.

### CODE REVIEW

**Comment by HugoDelatte** [comment]
Thanks for the PR.
Correct me if I'm wrong, but .cursor/mcp.json file is for direct local dev of skfolio so I don't think we should ship it in the library repo.

What we probably want is better context for all users of skfolio, for example using this: https://github.com/upstash/context7/blob/master/docs/adding-projects.md#advanced-configuration-with-context7json

WDYT?

File: context7.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+    "$schema": "https://context7.com/schema/context7.json",
+    "projectTitle": "skfolio",
+    "description": "skfolio is a Python library for portfolio optimization and risk management built on top of scikit-learn.",
+
+    "folders": [
+      "docs", 
+      "src/skfolio"
+    ],
```
Comment (HugoDelatte) [inline context7.json]: Add also the examples:

```

  "folders": [
    "src/skfolio",
    "examples",
    "docs"
  ],
  ```

File: context7.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+       "pyproject.toml",
+       "SECURITY.md"
+    ],
+
+    "rules": [
+      "Input data (X) should be asset price returns, not prices.",
+      "Follows scikit-learn API: use .fit(X, y) to train and .predict(X) to get weights.",
+      "Ensure data is pre-processed and does not contain NaNs before fitting."
+    ]
+  }
```
Comment (HugoDelatte) [inline context7.json]: I would live the rules empty and let the LLM figure it out based on context

**Comment by HugoDelatte** [comment]
It seems context7 doesn't read python files, only .rst, .md, or .txt
So we are missing out on the examples and API as they are generated at build time and into HTML (which isn't optimal either anyway).

Will need to dig deeper to find a clean solution (like creating a dedicated contect7 branch and building API+examples into .txt).

Or maybe it's just easier to make our own context/MCP.

WDYT?

**Comment by CarloNicolini** [comment]
> It seems context7 doesn't read python files, only .rst, .md, or .txt So we are missing out on the examples and API as they are generated at build time and into HTML (which isn't optimal either anyway).
> 
> Will need to dig deeper to find a clean solution (like creating a dedicated contect7 branch and building API+examples into .txt).
> 
> Or maybe it's just easier to make our own context/MCP.
> 
> WDYT?

When using Cursor I find much easier to use the repomix tool https://repomix.com/

My cursor MCP configuration is:

```json
{
  "mcpServers": {
    "skfolio-source": {
      "command": "npx",
      "cwd": "/Users/myhome/workspace/skfolio/src/skfolio",
      "args": [
        "-y",
        "repomix",
        "--mcp"
      ]
    },
  }
}
```

clearly I have cloned the latest skfolio source code inside "/Users/myhome/workspace/skfolio/src/skfolio" and have repomix installed.

**Comment by moyuanchen** [comment]
> It seems context7 doesn't read python files, only .rst, .md, or .txt So we are missing out on the examples and API as they are generated at build time and into HTML (which isn't optimal either anyway).

Context7 doesn't ready python files and it is safe to exclude the src folder. However the code examples in HTML can be found by context7 from the website skfolio.org, which is also indexed.

> Or maybe it's just easier to make our own context/MCP.

Making our own scp means that we can have granular custom tools that structure the generated code in a reproducible way when the agent is given some task by the user. For example, setting up data normalization pipelines or always include a robust standardized backtest.

I think it will not be easier but allows more functionality.

**Comment by HugoDelatte** [comment]
@moyuanchen does PR https://github.com/skfolio/skfolio/pull/239 replace the need for this one?

**Comment by moyuanchen** [comment]
@HugoDelatte Yes, #239 optimizes the token footprint of skfolio for context7, among a number of other conventions for LLMs to ingest libraries. This PR can be closed.

The only part that would be lost would be the rules in the context7.json:
- Input data (X) should be asset price returns, not prices.
- Follows scikit-learn API: use .fit(X, y) to train and .predict(X) to get weights.
- Ensure data is pre-processed and does not contain NaNs before fitting.

This may be a good opportunity to move these into llms_txt_description in docs/conf.py. This is at the beginning of llms.txt where we describe what this library does and can provide any guidelines to LLMs.

Currently it says:

> Python library for portfolio optimization and risk management built on top of scikit-learn: build, fine-tune, cross-validate and stress-test portfolio models.

Happy to open a quick follow up PR for this.
